### PR TITLE
board: efm32pg_stk3402a: Add option to compile for EFM32JG12B

### PR DESCRIPTION
**Pull request by chrta**
This PR adds the Soc files for the SiLabs EFM32 Jade Gecko (EFM32JG12B). There is no distinct development boards that is populated with this SoC.
Since firmware compiled for the EFM32JG12B runs on the equivalent EFM32PG12B SoC, the board efm32pg_stk3402a is extended, to be able to compile a "Jade Gecko"-firmware for this board. This can be done by specifying the board `efm32pg_stk3402a_jg` instead of `efm32pg_stk3402a`, see contained documentation.

This PR depends on https://github.com/zephyrproject-rtos/hal_silabs/pull/3 that is already merged. 

See also https://lists.zephyrproject.org/g/devel/topic/60553822#6533

### CODE REVIEW

File: boards/arm/efm32pg_stk3402a/doc/index.rst
Comment (dbkinder) [inline boards/arm/efm32pg_stk3402a/doc/index.rst]: Change these last two sentences to:
```
The Pearl Gecko contains contains an ARM® Cortex®-M4F core, and the Jade Gecko
an ARM® Cortex®-M3 core.  Other features such as memory and peripherals are the same.
```

**Review by dbkinder — CHANGES_REQUESTED** [review]
one doc tweak, otherwise doc LGTM

File: boards/arm/efm32pg_stk3402a/doc/index.rst
Comment (chrta) [reply boards/arm/efm32pg_stk3402a/doc/index.rst]: done

**Review by dbkinder — APPROVED** [review]
+1 for doc changes, thanks.

**Comment by chrta** [comment]
@MaureenHelm, @galak or @ioannisg
Can someone have a look at zephyrproject-rtos/hal_silabs#3? After merging it i can update this pr so it builds.

**Comment by MaureenHelm** [comment]
> @MaureenHelm, @galak or @ioannisg
> Can someone have a look at [zephyrproject-rtos/hal_silabs#3](https://github.com/zephyrproject-rtos/hal_silabs/pull/3)? After merging it i can update this pr so it builds.

You can get it to build before merging the hal pr like this: 
```diff
diff --git a/west.yml b/west.yml
index 06ed9a56ca..c9cd6d6143 100644
--- a/west.yml
+++ b/west.yml
@@ -53,7 +53,7 @@ manifest:
       revision: 85302959c0c659311cf90ac51d133e5ce19c9288
       path: modules/hal/microchip
     - name: hal_silabs
-      revision: 9151e614c23997074acd1096a3e8a9e5c255d5b9
+      revision: pull/3/head
       path: modules/hal/silabs
     - name: hal_st
       revision: fa481784b3c49780f18d50bafe00390ccb62b2ec
```
See also: https://docs.zephyrproject.org/latest/guides/modules.html#changes-to-existing-modules

**Comment by chrta** [comment]
@MaureenHelm Thank you, i was not aware of this.

**Comment by chrta** [comment]
Rebased and added myself to CODEOWNERS.

**Comment by galak** [comment]
@chrta looks like you still need a CODEOWNERS update.

**Comment by chrta** [comment]
@galak Added myself to CODEOWNERS.

**Review by galak — APPROVED** [review]

**Comment by galak** [comment]
LGTM.  Marking DNM until we close on the HAL side update.

**Comment by galak** [comment]
Please update west.yml now that HAL changes are merged.

**Comment by chrta** [comment]
> Please update west.yml now that HAL changes are merged.

Done. Everything should be ready now.

**Pull request merged**
